### PR TITLE
feat: added expression support to text input in ios

### PR DESCRIPTION
**Pull request by victormr**
Added expression support to the TextInput widget

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: iOS/Sources/Beagle/Sources/Components/TextInput+Renderable/TextInput.swift [hunk 1]
```diff
@@ -20,30 +20,43 @@ import BeagleSchema
...
                                                        onFocus: onFocus),
                                           controller: renderer.controller)
+        
         textInputView.styleId = styleId
         renderer.observe(value, andUpdate: \.text, in: textInputView)
+        renderer.observe(placeholder, andUpdate: \.placeholder, in: textInputView)
+        renderer.observe(type, andUpdate: \.inputType, in: textInputView)
+        renderer.observe(disabled, andUpdateManyIn: textInputView) { disabled in
+            if let disabled = disabled {
+                textInputView.isEnabled = !disabled
```
Comment (theffc) [inline iOS/Sources/Beagle/Sources/Components/TextInput+Renderable/TextInput.swift:34]: these update functions that are different from the ones inside `setup` functions, are another sign of dyssynchrony between these and Model

File: iOS/Sources/Beagle/Sources/Components/TextInput+Renderable/TextInput.swift [hunk 2]
```diff
@@ -20,30 +20,43 @@ import BeagleSchema
...
     class TextInputView: UITextField, UITextFieldDelegate, InputValue, WidgetStateObservable, ValidationErrorListener {
         
         struct Model {
-            var value: Expression<String>?
-            var placeholder: Expression<String>?
-            var disabled: Expression<Bool>?
-            var readOnly: Expression<Bool>?
-            var type: Expression<TextInputType>?
-            var hidden: Expression<Bool>?
+            var value: String?
```
[the remark this answers is not in the document]
Comment (victormr) [reply iOS/Sources/Beagle/Sources/Components/TextInput+Renderable/TextInput.swift]: done

File: iOS/Sources/Beagle/Sources/Components/TextInput+Renderable/TextInput.swift (hunk 1 above)
Comment (victormr) [reply iOS/Sources/Beagle/Sources/Components/TextInput+Renderable/TextInput.swift:34]: done

**Pull request merged**